ref/java/src/main/java/org/bech32/SegwitAddress.java

```diff
@@ -0,0 +1,113 @@
+package org.bech32;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.commons.lang3.ArrayUtils;
+
+public class SegwitAddress {
+
+    public static Pair<Byte, byte[]> decode(String hrp, String addr) throws Exception {
+
+        Pair<byte[], byte[]> p = Bech32.bech32Decode(addr);
+
+        byte[] hrpgot = p.getLeft();
+        String hrpgotStr =  new String(hrpgot);
+        if(hrpgot == null || hrpgotStr == null)  {
+          return Pair.of(null, null);
+        }
+        if (!hrp.equals(hrpgotStr))    {
+            throw new Exception("mismatching bech32 human readeable part");
+        }
+        if (!hrpgotStr.equalsIgnoreCase("bc") && !hrpgotStr.equalsIgnoreCase("tb"))    {
```

Comment:
I would leave that up to the caller, who should know what hrp is expected anyway.

Reply:
Updated.